plugins/services/src/js/components/modals/CreateServiceModalForm.js

```diff
@@ -82,7 +82,22 @@ class CreateServiceModalForm extends Component {
     // Hint: When you add something to the state, make sure to update the
     //       shouldComponentUpdate function, since we are trying to reduce
     //       the number of updates as much as possible.
+    // In the Next line we are destructing the config to keep labels as it is and even keep labels with an empty value
+    let { serviceConfig } = ServiceUtil.getServiceJSON(this.props.service);
```

Comment:
I don't understand what happens here (next 14 lines). 

You put the lables into a constant, remove? the labels from the config and add them again after stripping the empty properties, but only if there were labels?
This whole segment is super confusing to me, I need to lookup `omit` and `CreateServiceModalFormUtil.stripEmptyProperties`, understand why it's there twice.

It seems like you "just" want to have labels in the result of `CreateServiceModalFormUtil.stripEmptyProperties` if it wasn't empty and this change seems far to complicated for that, at least to me. 

Am I missing some context? Can you refactor this?